runtime/syntax/python.vim

```diff
@@ -156,18 +156,25 @@ syn match   pythonEscape	"\\$"
 "
 " and so on, as specified in the 'Python Language Reference'.
 " https://docs.python.org/reference/lexical_analysis.html#numeric-literals
+"
```

Comment:
As mentioned in #7923 I'm already preparing a lot of other changes including this one.
The support for Python 2 has already been dropped for keywords and builtins in PR #7855 and I am dropping it in my changes for this part as well.
The only difference is that I now have to make a separate PR for just this change because of sudden urgency of underscores in numbers as evidenced by two PRs related to it in last 24 hours. :ili